transactions/transactor_test.go

```diff
@@ -396,30 +437,123 @@ func (s *TxQueueTestSuite) TestContractCreation() {
+		}
 	}
+	return finalReceipt
+}
 
+func (s *TxQueueTestSuite) approveSignRequests(selectedAccount *account.SelectedExtKey, totalTransactions int) {
 	go func() {
-		for i := 1000; i > 0; i-- {
+		numTransactions := 0
+		for {
+			if s.endWaitingGoRoutine() {
+				return
+			}
 			req := s.manager.pendingSignRequests.First()
+
 			if req == nil {
 				time.Sleep(time.Millisecond)
 			} else {
+				numTransactions++
+				if numTransactions == totalTransactions {
+					go func() {
+						// getFinalResult <- s.manager.pendingSignRequests.Wait(req.ID, time.Second*1)
+					}()
+				}
 				s.manager.pendingSignRequests.Approve(req.ID, "", s.defaultSignTxArgs(), simpleVerifyFunc(selectedAccount)) // nolint: errcheck
-				break
 			}
 		}
 	}()
+}
 
-	hash, err := s.manager.SendTransaction(context.Background(), tx)
-	s.NoError(err)
-	backend.Commit()
-	receipt, err := backend.TransactionReceipt(context.TODO(), hash)
+func (s *TxQueueTestSuite) TestEthTransferWithSimulatedBackend() {
+	backend, selectedAccount := getSimulatedBackend(s)
+	transactions := []SendTxArgs{
+		{
+			Value: (*hexutil.Big)(big.NewInt(10)),
+			From:  selectedAccount.Address,
+		},
+	}
+
+	s.setupWaiter()
+	s.approveSignRequests(selectedAccount, len(transactions))
+	s.sendTransactions(backend, transactions)
+	s.teardownWaiter()
+}
+
+func (s *TxQueueTestSuite) TestContractCreationWithSimulatedBackend() {
+	backend, selectedAccount := getSimulatedBackend(s)
+	transactions := []SendTxArgs{
+		{
+			Input: hexutil.Bytes(gethcommon.FromHex(contract.ENSBin)),
+			From:  selectedAccount.Address,
+		},
+	}
+
+	s.setupWaiter()
+	s.approveSignRequests(selectedAccount, len(transactions))
+	receipt := s.sendTransactions(backend, transactions)
+	s.Equal(crypto.CreateAddress(selectedAccount.Address, 0), receipt.ContractAddress)
+	s.teardownWaiter()
+}
+
+func (s *TxQueueTestSuite) TestContractStateTransferWithSimulatedBackend() {
+	backend, selectedAccount := getSimulatedBackend(s)
+
+	setOwnerMethodID := "0x5b0fc9"
+	// Give the node a value so that we are reassigning ownership of something
+	// other than the 0 node, which already belongs to the current owner
+	node := [32]byte{1}
+	ensParsed, err := abi.JSON(strings.NewReader(contract.ENSABI))
 	s.NoError(err)
-	s.Equal(crypto.CreateAddress(testaddr, 0), receipt.ContractAddress)
+	secondOwnerAddress := account.FromAddress(TestConfig.Account1.Address)
+	setOwnerMethodData, _ := ensParsed.Pack("setOwner", setOwnerMethodID, node, secondOwnerAddress)
+
+	ownerMethodID := "0x02571b"
+	getOwnerMethodData, _ := ensParsed.Pack("owner", ownerMethodID, node)
+
+	transactions := []SendTxArgs{
+		{
+			// Create contract
+			Input: hexutil.Bytes(gethcommon.FromHex(contract.ENSBin)),
+			From:  selectedAccount.Address,
+		},
+		{
+			// Change contract owner
+			Input: hexutil.Bytes(setOwnerMethodData),
+			From:  selectedAccount.Address,
+		},
+		{
+			// Check that the contract owner is correct
+			Input: hexutil.Bytes(getOwnerMethodData),
```

Comment:
getOwner is not a transaction but a call that reads from storage. it cant have a receipt